polars/polars-lazy/src/physical_plan/expressions/group_iter.rs

```diff
@@ -59,8 +60,14 @@ struct LitIter<'a> {
 }
 
 impl<'a> LitIter<'a> {
-    fn new(array: ArrayRef, len: usize, name: &str) -> Self {
-        let mut series_container = Box::pin(Series::try_from((name, array.clone())).unwrap());
+    fn new(array: ArrayRef, len: usize, logical: &DataType, name: &str) -> Self {
+        let mut series_container = Box::pin(
+            Series::try_from((name, array.clone()))
```

Comment:
Can you use `Series::from_chunks_and_dtype_unchecked` here? That should `tag` `new` as unsafe.